### PR TITLE
#74 - Criando endpoint de login de usuário.

### PR DESCRIPTION
**Pull request by DanielViniciusAlves**
Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce to <repo_name>?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] I have read the CONTRIBUTING.md doc.
- [ ] I have added necessary documentation (if appropriate).
- [ ] Any dependent changes have been merged and published in downstream modules.

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: api/controllers/login.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
+const db = require('../models/index');
+const jwt = require('jsonwebtoken');
+
+class LoginController {
+    constructor() { }
+
+    validateEmail(email)  {
```
Comment (luis-gustavo) [inline api/controllers/login.js]: A validação do email também já está sendo feita pelo sequelize. Não necessariamente precisamos tirar isso, mas está redundante

File: api/controllers/login.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    }
+
+    async login(req, res) {
+        const { password, email } = req.body;
+        if (!email || !password) {
+            return res.status(400).json({
+              message: "Email or Password not present",
+            })
+        }
+        const login = new LoginController();
```
Comment (luis-gustavo) [inline api/controllers/login.js]: Pq a login controller está se auto instanciando?

File: api/controllers/login.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        var validateEmail = login.validateEmail(email)
+        try {
+            if(validateEmail){
+                const user = await db.user.findOne({ where: { email: email } });
```
Comment (luis-gustavo) [inline api/controllers/login.js]: Dá pra passar uma chamada pro sequelize com email e senha, assim evita ter que checar se a senha é igual

File: api/controllers/login.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+                if (!user) {
+                  res.status(400).json({
+                    message: "Login not successful",
```
Comment (luis-gustavo) [inline api/controllers/login.js]: Criar um objeto/classe separado pra conter as mensagens de erro. Assim todos os endpoints ficam com as mensagens iguais

File: api/package.json [hunk 5]
```diff
@@ -21,7 +21,8 @@
     "morgan": "~1.9.1",
     "nodemon": "^2.0.19",
     "pg": "^8.7.3",
-    "sequelize": "^6.21.2",
+    "sequelize": "^6.21.3",
+    "sqlite3": "^5.0.10",
```
Comment (luis-gustavo) [inline api/package.json]: Algum motivo pra ter adicionado isso aqui?

**Review by luis-gustavo — CHANGES_REQUESTED** [review]
Apenas fazer as correções citadas, de resto está show

**Comment by DanielViniciusAlves** [comment]
Não criei a classe com as mensagens pois acredito que seria mais adequado abir uma issue nova para isso.

**Comment by DanielViniciusAlves** [comment]
Atualizei a branch e corrigi os conflitos que existiam.

**Pull request merged**